src/checkpoint/CheckpointV2.ts

```diff
@@ -0,0 +1,283 @@
+import * as Logger from '../Logger'
+import { config } from '../Config'
+import * as State from '../State'
+import * as CycleDB from '../dbstore/cycles'
+import * as ReceiptDB from '../dbstore/receipts'
+import * as OriginalTxDB from '../dbstore/originalTxsData'
+import * as Data from '../Data/Data'
+import * as Utils from '../Utils'
+import {
+    CheckpointStatus,
+    CheckpointStatusType,
+    CheckpointSyncStatus,
+    upsertCheckpointStatus,
+    updateCheckpointStatus,
+    getCheckpointStatus,
+    getFailedCheckpointStatuses,
+    getOldestPendingOrFailedCheckpointStatus,
+    getCheckpointSyncRange
+} from '../dbstore/checkpointStatus'
+import { CheckpointBucket, CheckpointType } from './CheckpointData'
+import { cycleCheckpointManager } from './CycleData'
+import { receiptCheckpointManager } from './ReceiptData'
+import { originalTxCheckpointManager } from './OriginalTxsData'
+
+/**
+ * Converts CheckpointType to CheckpointStatusType
+ * @param checkpointType The CheckpointType to convert
+ * @returns The corresponding CheckpointStatusType
+ */
+function mapCheckpointTypeToStatusType(checkpointType: CheckpointType): CheckpointStatusType {
+    switch (checkpointType) {
+        case CheckpointType.Cycle:
+            return CheckpointStatusType.CYCLE
+        case CheckpointType.Receipt:
+            return CheckpointStatusType.RECEIPT
+        case CheckpointType.OriginalTx:
+            return CheckpointStatusType.ORIGINAL_TX
+        default:
+            throw new Error(`Unknown checkpoint type: ${checkpointType}`)
+    }
+}
+
+/**
+ * Records the status of a checkpoint bucket when it ages out
+ * @param bucket The checkpoint bucket that is aging out
+ * @param checkpointType The type of checkpoint
+ */
+export async function recordCheckpointStatus(
+    bucket: CheckpointBucket<any>,
+    checkpointType: CheckpointType
+): Promise<void> {
+    try {
+        const cycle = parseInt(bucket.bucketID, 10)
+        if (isNaN(cycle)) {
+            Logger.mainLogger.error(`Could not parse cycle number from bucket ID: ${bucket.bucketID}`)
+            return
+        }
+
+        // Calculate if we have majority consensus
+        const totalArchivers = State.activeArchivers.length
+        const requiredMajority = Math.floor(totalArchivers / 2) + 1
+
+        // Check each radix entry for consensus
+        const failedRadixes: string[] = []
+        let matchedArchivers = 0
+
+        // Count how many radixes have majority consensus
+        let radixesWithMajority = 0
+
+        for (const [radix, entry] of bucket.peerRadixDigests.entries()) {
+            let maxVotes = 0
+            let totalVotes = 0
+
+            // Find the hash with the most votes
+            for (const [hash, votes] of entry.hashTally.entries()) {
+                totalVotes += votes
+                if (votes > maxVotes) {
```

Comment:
should this be ? If not, then `totalVotes` is not needed as its not used anywhere else 
```suggestion
                if (totalVotes > maxVotes) {
```